assets/aws/files/install-hardened.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+if [[ "${DEBUG:-false}" == "true" ]]; then
+    set -x
+fi
+
+# Set some curl options so that temporary failures get retried
+# More info: https://ec.haxx.se/usingcurl-timeouts.html
+CURL_OPTS="-L --retry 100 --retry-delay 0 --connect-timeout 10 --max-time 300"
+
+# Update packages
+dnf -y update
+
+# Install 
+#  - uuid used for random token generation,
+#  - python for certbot
+dnf install -y uuid python3
+
+# Install certbot 
+python3 -m venv /opt/certbot
+/opt/certbot/bin/pip install --upgrade pip
+/opt/certbot/bin/pip install certbot certbot-dns-route53
+ln -s /opt/certbot/bin/certbot /usr/bin/certbot
+
+# Create teleport user. It is helpful to share the same UID
+# to have the same permissions on shared NFS volumes across auth servers and for consistency.
+useradd -r teleport -u ${TELEPORT_UID} -d /var/lib/teleport
+# Add teleport to adm group to read and write logs
+usermod -a -G adm teleport
+
+# Setup teleport run dir for pid files
+mkdir -p /run/teleport/ /var/lib/teleport /etc/teleport.d
+chmod 0700 /var/lib/teleport
+chown -R teleport:adm /run/teleport /var/lib/teleport /etc/teleport.d/
```

Comment:
Not sure that `chown -R` is needed if you have created empty directories. Perhaps use `install(1)`?

```
install -d -m 0700 -o teleport -g adm /var/lib/teleport
install -d -m 0755 -o teleport -g adm /run/teleport /etc/teleport.d
```